devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -435,6 +436,19 @@ bool HumanStateProvider::open(yarp::os::Searchable& config)
         }
     }
 
+     if (!(config.check("CalibrationConfig") && config.find("CalibrationConfig").isList())) {
+        yInfo() << LogPrefix << "CalibrationConfig option not found or not valid, all the model joints are selected.";
+        pImpl->CalibrationConfig.clear();
+    }
+    else
+    {
+        auto CalibrationConfigBottle = config.find("CalibrationConfig").asList();
+        for (size_t it = 0; it < CalibrationConfigBottle->size(); it++)
```

Comment:
Here you should check if the size of the bottle is the same size of the joint list

Reply:
Same as above. I'll steal some of your time on the next days (or maybe @RiccardoGrieco's time since he likes to heart this pr :) )

Reply:
Sure, no problem!

Reply:
Solved in https://github.com/robotology/human-dynamics-estimation/pull/341/commits/e59e7907e431dbdd0950bfd52ac89509d1abe1f0

Reply:
@lrapetti if you are ok with the changes introduced with https://github.com/robotology/human-dynamics-estimation/commit/e59e7907e431dbdd0950bfd52ac89509d1abe1f0, please resolve this conversation.

Reply:
Looks good, thanks!